### DISCUSSION TITLE
Sponsor images are looking broken on scheduled action runs

### DISCUSSION THREAD

**Discussion by K-vanc** [JamesIves/github-sponsors-readme-action#882] [Q&A] [answered]
Hi and thanks for good work.

I am not sure if it is a bug or something related with Github but everything was running fine for long time and last week I noticed sponsor images in a dedicated tier were displayed broken so I rerun the action and they were fixed. I thought it was some temporarily problem of github similar like sometimes unresponsive download numbers etc. but I noticed that it is only happening when action run as "scheduled".

I compared deploy log of scheduled run and my re-runs but there is no difference except some changed hash values. There is also no error reported. Urls are looking ok as well but it is diplayed as below on scheduled;

![image](https://github.com/user-attachments/assets/1c80c543-9f6c-4ec0-9f89-18afc675782a)

and it is becoming normal if i rerun action till next scheduled run. I checked also in different browsers to understand if it is some kind of html  rendering issue but displayed as on photo.

I hope this info is sufficent for you since I have almost 0 knowledge about actions or maybe someone else is experiencing same issue

**Comment by K-vanc** [comment]
now I see there is an extra parameter for "permission" on your read.me file which is missing on my .yml file. I was thinking it could be some kind of permission issue so maybe this could make the trick. 

I will test and write here back

**Reply by JamesIves** [reply]
Any luck? I took a look at your workflow file and everything looks right to me.

**Comment by K-vanc** [comment]
Hi again. Yes, I lost a bit time for activating change on .yml file but I can confirm adding below to .yml file fixed it;

`permissions:
  contents: write`

Thanks again for your check

**Comment by K-vanc** [comment]
Hi again,

It seems like images(probably tokens inside their url) are expiring after 1 or 2 hours and images becoming broken. Yesterday I set my cron to every 5 mins to test if scheduled runs will work then after confirming there is no problem, set it back to regular 8 hours run but noticed they were broken again yesterday night. This morning I started investigating abit more by html decoding working and non-working images and trying to open them in browser. After a few hours (1 or 2, I was busy so dont know exactly when happened), working images not opened in browser when I refresh and at the same they were also broken again in readme.md without any new scheduled or manual action run.

I red somewhere here as you made some changes for using private avatar images but do you think they may have some time limit for token expiration?

**Comment by K-vanc** [comment]
I think same thing happening to your readme.md as well if you check below snapshot for your avatars;

![image](https://github.com/user-attachments/assets/ea281c14-3238-4ae5-9f0c-184c56c0115c)

I am pretty sure that avatars were visible this morning

**Comment by K-vanc** [comment, marked as answer]
I think i am correct about that expire date. I rerun action and copied jwt token part of refreshed avatar url then decrypted jwt token. It has below data;

```
{
  "iss": "github.com",
  "aud": "raw.githubusercontent.com",
  "key": "key1",
  "exp": 1734435600,
  "nbf": 1734434400,
  "path": "/u/XXXXXXX"
}
```

I modified path value since it is the only part containing private data. Anyway expire timestamp shows this image will die at 17.12.2024 14:40 +0300 which is a few minutes later. Currently image is active both in readme.md and browser. I will update result 5 mins later

Edit:

Image url died as soon as reached timestamp in exp value. It is still taking a little bit more time for .md file update but url is not reacheable anymore. I hope this helps about problem

**Reply by JamesIves** [reply]
I'll take a look- not sure why the api is providing links to images that expire. I'll try and patch it before my flight later today.

**Reply by JamesIves** [reply]
Pushed up a change for this, it now relies on the always accessible github.com/{username}.png avatar urls - ie https://github.com/K-vanc.png

Available in 1.5.6

**Reply by K-vanc** [reply]
It works as usual now. Thanks for your prompt action. Have a good flight(and probably a vacation)
